src/llm/tokenizer.py

```diff
@@ -0,0 +1,127 @@
+from typing import Union
+
+from tokenizers import (
+    Tokenizer,
+    models,
+    normalizers,
+    pre_tokenizers,
+    decoders,
+    trainers,
+)
+from datasets import load_dataset
+
+
+def get_dataset(name):
+    # Load the dataset
+    train_dataset = load_dataset("wmt/wmt14", name=name, split="train")
+    val_dataset = load_dataset("wmt/wmt14", name=name, split="validation")
+    test_dataset = load_dataset("wmt/wmt14", name=name, split="test")
+
+    return {
+        "train": train_dataset,
+        "val": val_dataset,
+        "test": test_dataset,
+    }
+
+
+def get_tokenizers(dataset_name, path="data"):
+    """
+    Get the tokenizers for the specified dataset name.
+    """
+    try:
+        tokenizers = {}
+        for lang in dataset_name.split("-"):
+            tokenizers[lang] = Tokenizer.from_file(f"{path}/{lang}_tokenizer.json")
+        return tokenizers
+    except FileNotFoundError:
+        raise FileNotFoundError(
+            f"Tokenizers for {dataset_name} not found. Please train the tokenizers first using python src/llm/train_tokenizer.py."
+        )
+
+
+def decode_tokens(tokenizer, tokens: Union[list[int], list[list[int]]], batch=False):
+    if batch:
+        return tokenizer.decode_batch(tokens)
+    else:
+        return tokenizer.decode(tokens)
```

Comment:
Can we infer the `batch` flag from the `tokens` parameter?

Reply:
I have marked this as a TODO for now. Will do it when we get to it. For now it is just a utility that I used to see if the tokenizer is working fine.